app/uk/gov/hmrc/fileupload/microserviceGlobal.scala

```diff
@@ -186,87 +219,71 @@ object MicroserviceGlobal extends DefaultMicroserviceGlobal with RunMode {
     new RoutingController(envelopeCommandHandler)
   }
 
-  def basicAuthConfiguration(config: Configuration): BasicAuthConfiguration = {
-    def getUsers(config: Configuration): List[User] = {
-      config.getString("basicAuth.authorizedUsers").map { s =>
-        s.split(";").flatMap(
-          user => {
-            user.split(":") match {
-              case Array(username, password) => Some(User(username, password))
-              case _ => None
-            }
-          }
-        ).toList
-      }.getOrElse(List.empty)
+  lazy val appRoutes = new AppRoutes(httpErrorHandler, new Provider[EnvelopeController] {
+    override def get(): EnvelopeController = envelopeController
+  },
+    new Provider[FileController] {
+      override def get(): FileController = fileController
+    }, new Provider[EventController] {
+      override def get(): EventController = eventController
+    }, new Provider[CommandController] {
+      override def get(): CommandController = commandController
     }
+  )
 
-    config.getBoolean("feature.basicAuthEnabled").getOrElse(false) match {
-      case true => BasicAuthEnabled(getUsers(config))
-      case false => BasicAuthDisabled
-    }
-  }
+  lazy val transferRoutes = new TransferRoutes(httpErrorHandler, new Provider[TransferController] {
+    override def get(): TransferController = transferController
+  })
+  lazy val routingRoutes = new RoutingRoutes(httpErrorHandler, new Provider[RoutingController] {
+    override def get(): RoutingController = routingController
+  })
+  lazy val metricsController = new MetricsController(new GraphiteMetricsImpl(applicationLifecycle, configuration))
+  lazy val adminRoutes = new AdminRoutes(httpErrorHandler, new Provider[MetricsController] {
+    override def get(): MetricsController = metricsController
+  })
 
-  override def onStart(app: Application): Unit = {
-    super.onStart(app)
+  lazy val router: Router = new Routes(httpErrorHandler, appRoutes, transferRoutes, routingRoutes,
+    health.Routes, adminRoutes)
```

Comment:
yes, I think we somehow overlooked TestOnlyController when we first started refactoring the code to deal with that later. I'll add it to the mix. Are you aware of any flag or env variable for enabling/disabling test only endpoints though?